### ISSUE TITLE
[D3D12-ROV] Xenia Closes without Warning

### ISSUE THREAD

**Issue by escape209** [xenia-project/xenia#1251]
[//]: # (Describe what's going wrong:)
With ROV enabled in D3D12, Xenia may freeze for several seconds before randomly closing without any warning or error messages. I was able to produce this in Burnout Revenge by attempting to access the garage.

[//]: # (Describe what should happen:)
Xenia shouldn't just close, an error message should appear before this happens.

[//]: # (If applicable, provide a callstack here - esp. for crashes)

[//]: # (If applicable, upload a logfile and link it here)
[xenia.log](https://github.com/xenia-project/xenia/files/2545045/xenia.log)

**Comment by win98and95** [comment]
Sometimes the error messages doesn't even appears it just freezes

**Comment by LuismaSP89** [comment]
Update: Xenia ROV 8f3d5a2c Nov 14-2018 crashes immediately in eternal sonata (Graphic device lost)

**Comment by LuismaSP89** [comment]
@Eurose209 Update: For me, in latest build c2786eef Nov 16 2018 all problems with ROV are solved, you sould retest :P

**Comment by Eurose209** [comment]
@LuismaSP89 Still experiencing the issue with that commit, unfortunately.

**Comment by tetration** [comment]
> @LuismaSP89 Still experiencing the issue with that commit, unfortunately.

Same here. Since this commit I have been experiencing more crashes with the "graphics device lost"

**Comment by niko1point0** [comment]
Hey everyone, I had the same issue. Try checking your GPU Memory Usage. For me, I got that error when all of my GPU Memory was filled, and all of my CPU RAM was full. Honestly I'm not sure if the DirectX 12 ROV ever deletes memory after it is allocated

**Comment by niko1point0** [comment]
Hey guys, this issue will be fixed very soon, I got a message from the Lead Developer of the D3D12 ROV

Triang3l says:
"Currently it never deallocates textures, but I'm planning to add least-recently-used caching."

That is why we are having memory leaks, and that is why everyone is getting the "graphics device lost" error message

Don't worry, he says that he is working on fixing the issue

**Comment by Triang3l** [comment]
It's a GPU hang caused by something wrong happening in shaders, not a memory allocation failure. It often happens even when starting some games, and also the RTV/DSV renderer uses even more memory, but device losses happen more often in the ROV renderer.

**Comment by Eurose209** [comment]
This can be closed as it has been fixed as of Radeon Software 18.12.2.

**Comment by lif3mau5** [comment]
No it's not fixed yet!

**Comment by chris-hawley** [comment]
> No it's not fixed yet!

care to elaborate @lif3mau5 ?

**Comment by Triang3l** [comment]
AMD's driver has a relatively unstable shader compiler with null pointer accesses that are very hard to debug, I have lots of crashes on Vega 10.

**Comment by byzyn4ik** [comment]
Can't start many games in ROV mode with vega 64. Only Diablo 3 is running but p[erformance is lower than without ROV. Tried different drivers.

**Comment by Parovozik** [comment]
Hello. On all new versions of Xenia i get crash on start any game via Dx12-ROV, but it works via Dx12-RTV and Vulkan (ROV only on old versions of Xenia). As if support for dx12 extensions is not recognized, but it worked before. I tried some versions of GPU driver until 431.60 and even reinstall Windows (my gpu is gtx1070). Any suggestions?
Crash Logs:
[xenia.log](https://github.com/xenia-project/xenia/files/3468133/xenia.log)
[xeniaa.log](https://github.com/xenia-project/xenia/files/3468134/xeniaa.log)
[xeniae.log](https://github.com/xenia-project/xenia/files/3468136/xeniae.log)

**Comment by illusion98** [comment]
THIS IS NOT A SUPPORT FORUM! For support go here:
Xenia Discord (#help) - https://discord.me/xenia-emulator

**Comment by sanjaysatoji** [comment]
I'm still getting crashes with ROV enabled on 19.9.2 on an AMD Radeon VII on the latest xenia builds

**Comment by Salvo8110** [comment]
Good morning I also have crashes after the intro with amd Radeon adrenalin Edition 20.1.4 with d3d12 rov active. my pc is ryzen 5 3500 with vega 8. Are there any news? Thank you

**Comment by Margen67** [comment]
@Salvo8110 Here is the news: Don't spam GitHub, and wait for the issue to be fixed.

**Comment by Nikhil111777** [comment]
Disabling the d3dd12 rov also not working on my i5 amd 20.1.4

**Comment by Triang3l** [comment]
Reposting from a semi-duplicate issue #1763 by @AD010, similar CreateGraphicsPipelineState crashes also happen on Intel HD GT1 (Haswell):

> 
> 
> The AVX part is duplicate of #603, and it won't be fixed any time soon — requires changes in around 600 places in the code, and hard to test (because if we miss some instruction, it will still work on developers' hardware).
> 
> For GT1, reproduction info is needed — the name of the game, steps to do in the game to locate a place using the shader that causes the crash (or the shader cache file for that game). I do, however, remember some pipeline state objects failing to be created on Intel HD Graphics 4400 at least in Halo 3.

The Incredible Hulk (2008) and Vigilante 8 Arcade, p.s. on RPCS3 all games (on all settings) have black or white screen without sounds, my Intel HD have full support OpenGL 1-4.3
[shareable.zip](https://github.com/xenia-project/xenia/files/6049491/shareable.zip)
from old version : 
[shareable.zip](https://github.com/xenia-project/xenia/files/6049543/shareable.zip)

**Comment by mirh** [comment]
Opengl has nothing to do with xenia
*Anyhow* I can also report something similar when I use my HD 520. 
Somehow the first time I started xenia I could get into the menu, but after that now it's a hang while "preloading shaders".

**Comment by harrishandoko** [comment]
Hi Guys, I'm experiencing a consistent crashing from an xbox 360 "Rumble Roses XX" game I'm trying on Xenia. Managed to get it running, but right at the start of a match, it crashes without warning. No error messages or anything. I have the xenia.log attached here. My PC specs is AMD Ryzen 5 3600, with NVidia RTX 2070, 16 GB RAM. Before I extracted the .xex file from the ISO, the game was actually crashing even at the character selection. Can anyone give me some hints on what could be the issue here?
Thanks very much.

[xenia.log](https://github.com/xenia-project/xenia/files/6966120/xenia.log)
